File: server/src/main/java/pt/iade/ei/thinktoilet/models/entities/Extra.java
package pt.iade.ei.thinktoilet.models.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name = "extra")
public class Extra {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "extra_id")
    private int id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
            name = "extra_toil_id",
            referencedColumnName = "toil_id",
            nullable = false
    )
    private Toilet toilet;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
            name = "extra_tex_id",
            referencedColumnName = "tex_id",
            nullable = false
    )
    private TypeExtra typeExtra;
}

File: app/app/src/main/java/pt/iade/ei/thinktoilet/models/Extra.kt
package pt.iade.ei.thinktoilet.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Extra(
    @SerializedName("id") val id: Int?,
) : Serializable

object TypeExtra {
    const val ACCESSIBILITY = 1
    const val BABY_CHANGING_STATION = 2
}